app/main.py

```
@@ -1,2 +1,123 @@
+                index = hash(old_hash_table[idx][0]) % self.length
+                for i in range(len(self.hash_table)):
+                    s_index = (index + i) % self.length
+                    if self.hash_table[s_index] is None:
+                        self.hash_table[s_index] = old_hash_table[idx]
+                        break
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        if self.capacity / self.length >= (2 / 3):
+            self._resize()
+        index = hash(key) % self.length
+        for i in range(self.length):
+            s_index = (index + i) % self.length
+            if self.hash_table[s_index] is None:
+                self.capacity += 1
+                self.hash_table[s_index] = [key, hash(key), value]
+                break
+            elif (self.hash_table[s_index][1] == hash(key)
+                  and self.hash_table[s_index][0] == key):
+                self.hash_table[s_index][2] = value
+                break
+
+    def __getitem__(self, key: Any) -> Any:
+        index = hash(key) % self.length
+        for i in range(self.length):
+            s_index = (index + i) % self.length
+            if self.hash_table[s_index] is None:
+                raise KeyError("Key not found")
+            elif self.hash_table[s_index][0] == key:
+                return self.hash_table[s_index][2]
+        raise KeyError("Key not found")
+
+    def __len__(self) -> int:
+        return self.capacity
+
+    def __delitem__(self, key: Any) -> None:
+        index = hash(key) % self.length
+        for i in range(self.length):
+            s_index = (index + i) % self.length
+            if self.hash_table[s_index] is None:
+                return
+            elif (self.hash_table[s_index][1] == hash(key)
+                  and self.hash_table[s_index][0] == key):
+                self.capacity -= 1
+                self.hash_table[s_index] = None
+                self._recreate()
+                break
```

`__delitem__` currently returns silently if it encounters a None slot in probing (lines 102-103). While the checklist notes `Note: __delitem__ should return nothing` (so successful deletion must not return a value), Python `dict.__delitem__` raises `KeyError` when the key is missing. For correct dict-like semantics and clearer API behavior, raise a `KeyError` with a message when the key is not found instead of returning silently. Also ensure capacity is decremented only when an actual entry is removed (you already decrement on removal). Quote from checklist about deletion test presence: `If you want check your __delitem__ method uncomment test_deletion():` .